uclapi.openapi.json

```diff
@@ -0,0 +1,4256 @@
+          }
+        }
+      }
+    },
+    "/roombookings/freerooms" : {
+      "get" : {
+        "summary" : "Given a start time and an end time, this endpoint returns all rooms which are free in that time range.",
+        "description": "Note: This endpoint only returns publicly bookable rooms. Departmentally bookable rooms are not included.",
+        "tags" : [ "Room Bookings" ],
+        "security" : [ {
+          "OAuthSecurity" : [ ],
+          "OAuthToken" : [ ]
+        } ],
+        "parameters" : [ {
+          "$ref" : "#/components/parameters/start_datetime"
+        }, {
+          "$ref" : "#/components/parameters/end_datetime"
+        } ],
+        "responses" : {
+          "200" : {
+            "description" : "Success",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                  "type" : "object",
+                  "properties" : {
+                    "ok" : {
+                      "type" : "boolean"
+                    },
+                    "count" : {
+                      "type" : "integer"
+                    },
+                    "free_rooms" : {
+                      "type" : "array",
+                      "items" : {
+                        "$ref" : "#/components/schemas/room"
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "400" : {
+            "description" : "Request error",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                  "$ref" : "#/components/schemas/error"
+                },
+                "examples" : {
+                  "1" : {
+                    "$ref" : "#/components/examples/ErrorNoToken"
+                  },
+                  "2" : {
+                    "$ref" : "#/components/examples/ErrorTokenNotExist"
+                  },
+                  "3" : {
+                    "$ref" : "#/components/examples/ErrorInvalidToken"
+                  },
+                  "4" : {
+                    "$ref" : "#/components/examples/ErrorThrottling"
+                  },
+                  "5" : {
+                    "$ref" : "#/components/examples/ErrorNoClientSecret"
+                  },
+                  "6" : {
+                    "$ref" : "#/components/examples/ErrorInvalidClientSecret"
+                  },
+                  "7" : {
+                    "$ref" : "#/components/examples/ErrorInactiveToken"
+                  },
+                  "Invalid parameters" : {
+                    "value" : {
+                      "ok" : false,
+                      "error" : "start_datetime or end_datetime not provided"
+                    }
+                  },
+                  "Invalid date/time" : {
+                    "value" : {
+                      "ok" : false,
+                      "error" : "date/time isn't formatted as suggested in the docs"
```

Comment:
nit: I think this is a mistake in the API, I believe we format date/time as datetime

Reply:
updated error message to be more clear: `start_datetime/end_datetime isn't formatted as suggested in the docs`